pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -77,4 +80,68 @@ public static boolean arrayContainsInt(int[] values, int valueToFind) {
   public static boolean arrayContainsString(String[] values, String valueToFind) {
     return ArrayUtils.contains(values, valueToFind);
   }
+
+  @ScalarFunction
+  public static String[] arraySliceString(String[] values, int start, int end) {
+    return Arrays.copyOfRange(values, start, end);
+  }
+
+  @ScalarFunction
+  public static int[] arraySliceInt(int[] values, int start, int end) {
+    return Arrays.copyOfRange(values, start, end);
+  }
+
+  @ScalarFunction
+  public static int[] arrayDistinctInt(int[] values) {
+    return Arrays.stream(values).distinct().toArray();
```

Comment:
Try to avoid using stream api in query execution code path because its performance is much worth than regular for loop